modules/commons/src/main/java/org/apache/synapse/commons/util/FilePropertyLoader.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.synapse.commons.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ *  File Property loader can be used to load the file property variables.
+ */
+public class FilePropertyLoader {
+
+    private static final Log LOG = LogFactory.getLog(FilePropertyLoader.class);
+    private static final String CONF_LOCATION = "conf.location";
+    private static final String SYNAPSE_PROPERTY_FILE = "synapse.properties";
+    private static final String FILE_PROPERTY_PATH = "synapse.commons.file.properties.location";
+    private static final String FILE_PROPERTY_FILENAME = "synapse.commons.file.properties.file.name";
+    private static FilePropertyLoader fileLoaderInstance = new FilePropertyLoader();
+    private static Map fileProperty;
+    private static String fileValue;
+
+    private FilePropertyLoader(){}
+
+    public static FilePropertyLoader getFileLoaderInstance(){
+        return fileLoaderInstance;
+    }
+
+    public static Map getFileProperty() {
+        return fileProperty;
+    }
+
+    public static void setFileProperty(Map fileProperty) {
+        FilePropertyLoader.fileProperty = fileProperty;
+    }
+
+    public static String getFileValue() {
+        return fileValue;
+    }
+
+    public static void setFileValue(String input) {
+        FilePropertyLoader.fileValue = (String) getFileProperty().get(input);
+    }
+
+    public static void loadPropertiesFile(){
+        try {
+            fileProperty = readFileProperties();
+
+            if (fileProperty != null) {
+                setFileProperty(fileProperty);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Let's log the error.

Reply:
fix with the commit https://github.com/wso2/wso2-synapse/pull/1431/commits/e2f0dae8565f69847833751ae15b2db323f42b4d